packages/qwik/src/qwikloader.ts

```diff
@@ -32,8 +35,20 @@ export const qwikLoader = (
   const isConnected = 'isConnected';
   const qvisible = 'qvisible';
   const Q_JSON = '_qwikjson_';
+  const qContainerAttr = '[q\\:container]';
+  const nativeQuerySelectorAll = (root: ParentNode, selector: string) =>
+    Array.from(root.querySelectorAll(selector));
   const querySelectorAll = (query: string) => {
-    return doc.querySelectorAll(query);
+    const elements: Element[] = [];
+    roots.forEach((root) => elements.push(...nativeQuerySelectorAll(root, query)));
```

Comment:
since they're all roots and assumed to be independant the order of roots doesn't matter right?

Reply:
Correct. What are you getting at?